### ISSUE TITLE
Using the API with Authentication

### ISSUE THREAD

**Issue by taf2** [googleapis/nodejs-speech#69]
Browsing samples authentication appears to be ignored.   So we get errors like the one below:

```
Error: Unexpected error while acquiring application default credentials: Could not load the default credentials. Browse to https://developers.google.com/accounts/docs/application-default-credentials for more information.
    at GoogleAuth.<anonymous> (/Users/taf2/work/ctm-ai/node_modules/google-auth-library/build/src/auth/googleauth.js:235:31)
    at step (/Users/taf2/work/ctm-ai/node_modules/google-auth-library/build/src/auth/googleauth.js:47:23)
    at Object.next (/Users/taf2/work/ctm-ai/node_modules/google-auth-library/build/src/auth/googleauth.js:28:53)
    at fulfilled (/Users/taf2/work/ctm-ai/node_modules/google-auth-library/build/src/auth/googleauth.js:19:58)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

It might be nice to include examples that show how to exactly pass the API keys into the code so people don't have to guess... 

```

Before you begin
Select or create a Cloud Platform project.

Go to the projects page

Enable billing for your project.

Enable billing

Enable the Google Cloud Speech API API.

Enable the API

Set up authentication with a service account so you can access the API from your local workstation.
```

from the README - takes me to pages that don't explain how to use the node.js API to authenticate.... 

Contrast that with examples from similar competing projects:

https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/TranscribeService.html
http://voicebase.readthedocs.io/en/v3/how-to-guides/hello-world.html#how-to-get-your-bearer-token
https://console.bluemix.net/docs/services/watson/getting-started-tokens.html#tokens-for-authentication

I just think a few one liners, making this very obvious would help more people use this API with less friction to getting started...

this page for example doesn't help:
https://cloud.google.com/docs/authentication/getting-started#auth-cloud-implicit-nodejs
```
// Imports the Google Cloud client library.
const Storage = require('@google-cloud/storage');

// Instantiates a client. If you don't specify credentials when constructing
// the client, the client library will look for credentials in the
// environment.
const storage = new Storage();

// Makes an authenticated API request.
storage
  .getBuckets()
  .then((results) => {
    const buckets = results[0];

    console.log('Buckets:');
    buckets.forEach((bucket) => {
      console.log(bucket.name);
    });
  })
  .catch((err) => {
    console.error('ERROR:', err);
  });```

I don't see any thing in there about where to place an API key?

**Comment by phsultan** [comment]
I guess the most common way to use this package is to authenticate using a service account and set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable to point to the JSON file that matches with the created service account. The token retrieval/refresh procedures are all handled by underlying (gRPC and authentication) packages.

This authentication procedure applies to quite a few other Google services, most probably those which are gRPC based if I'm not mistaken.

Now it looks like authenticating using an API key isn't that easy. You might want to take a look at that issue : https://github.com/googleapis/nodejs-speech/issues/19 that contains some examples of how to use the API the 'REST' way (e.g. with `axios`) without using `nodejs-speech`.

Hope this helps!

**Comment by taf2** [comment]
Definitely, I just think the readme's on most of the API language bindings should provide a bit more of a complete example... while it's nice that most if not all google API's authenticate in a similar if not unified way... because it is not similar to really any other services... it would or seem to make sense that more language specific examples existed... because without that virtually all samples provided are broken.

**Comment by stephenplusplus** [comment]
This issue was moved to https://github.com/googleapis/gapic-generator/issues/1959.

**Comment by stephenplusplus** [comment]
I agree @taf2. I moved the issue to the repo that generates the READMEs for all of our repos.

**Comment by stephenplusplus** [comment]
> this page for example doesn't help:
https://cloud.google.com/docs/authentication/getting-started#auth-cloud-implicit-nodejs

I actually found that example to be helpful. Directly above the sample is a section called "Verifying authentication". The first link takes you to a page that has many details and steps that show how to authenticate with both an environment variable and a service account JSON keyfile: https://cloud.google.com/docs/authentication/production#setting_the_environment_variable

---

From https://github.com/googleapis/gapic-generator/issues/1959#issuecomment-385989042:

> My original point was the documentation in general for this API fails to make it easy for new people starting out to quickly get started... Closing this out doesn't really change that right?

I think changing the README, a common first place to look for help with a module, would help the most. If you have any specific suggestions of things that can be fixed, and I'm not picking up on them, a PR would be very welcome.

**Comment by jackiehluo** [comment]
+1, I found this process super, super confusing. And agreed that the example for the Node SDK explains that the credentials are implicitly detected, but there are no examples I found that explain how you're supposed to include them in your project. It's taken me hours to just get to this issue, which is the first one that's indicated that I'm supposed to point the env variable to the service account JSON file.
